spec/defines/dns__server__options_spec.rb

```diff
@@ -70,6 +70,26 @@
     it { should raise_error(Puppet::Error, /is not an Array/) }
   end
 
+  context 'when passing valid array to listen_on_ipv6' do
+    let :params do
+      { :listen_on_ipv6 => [ '2001:db8:1::1', '2001:db8:2::/124' ] }
+    end
+    it { should contain_file('/etc/bind/named.conf.options').with_content(/2001:db8:1::1;$/)  }
+    it { should contain_file('/etc/bind/named.conf.options').with_content(/2001:db8:2::\/124;$/)  }
+  end
+
+  context 'when passing a string to listen_on_ipv6' do
```

Comment:
I don't think you realy have to double check the validate_array function, it will be fine